sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/FileTableSuite.scala

```diff
@@ -85,4 +86,35 @@ class FileTableSuite extends QueryTest with SharedSparkSession {
       assert(table.dataSchema == expectedDataSchema)
     }
   }
+
+  test("SPARK-30289: DSv2 `FileTable` partitioning should not accept nested columns") {
+    val nestedSchema = StructType(Array(
+      StructField("nested", StructType(Array(
+        StructField("id", IntegerType, true),
+        StructField("data", StringType, true)
+      )), true)
+    ))
+
+    val e = intercept[AnalysisException] {
+      new FileTable(spark, CaseInsensitiveStringMap.empty(), Seq(), Some(nestedSchema)) {
+        override def partitioning: Array[Transform] =
+          Array(IdentityTransform(FieldReference(Seq("nested", "id"))))
```

Comment:
Shall we update like the following? Then, we can remove the rest (line 102 ~ 115).
```scala
- new FileTable(spark, CaseInsensitiveStringMap.empty(), Seq(), Some(nestedSchema)) {
+ new DummyFileTable(spark, CaseInsensitiveStringMap.empty(), Seq(), nestedSchema, None) {
    override def partitioning: Array[Transform] =
      Array(IdentityTransform(FieldReference(Seq("nested", "id"))))
```